psycop/common/model_training_v2/classifier_pipelines/binary_classification_pipeline.py

```diff
@@ -10,10 +11,19 @@
 
 class BinaryClassificationPipeline:
     def __init__(self, steps: Sequence[ModelStep]):
-        ...
+        self.pipe = Pipeline(steps=steps)
 
-    def fit(self, X: PolarsFrame, y: PolarsFrame) -> None:
-        ...
+    def fit(self, X: PolarsFrame, y: pl.Series) -> None:
+        if isinstance(X, pl.LazyFrame):
+            # pyright thinks "X" is a constant since it's all caps
+            # hence we type ignore
+            X = X.collect()  # type: ignore
+        self.pipe.fit(X.to_pandas(), y)
 
     def predict_proba(self, X: PolarsFrame) -> pl.Series:
-        ...
+        """Returns the predicted probabilities of the `1`
+        class"""
+        if isinstance(X, pl.LazyFrame):
+            X = X.collect()  # type: ignore
+        pred_probs = self.pipe.predict_proba(X.to_pandas())[:, 1]
+        return pl.Series("y_hat_probs", pred_probs)
```

Comment:
Would be neat if `y_hat_probs` this could be type-hinted in the return type. E.g. the return type is a "PredProbaSeries" or similar.